dsc_lib/src/functions/join.rs

```diff
@@ -11,14 +11,14 @@ use tracing::debug;
 #[derive(Debug, Default)]
 pub struct Join {}
 
-fn stringify_value(v: &Value) -> String {
+fn stringify_value(v: &Value) -> Result<String, DscError> {
     match v {
-        Value::String(s) => s.clone(),
-        Value::Number(n) => n.to_string(),
-        Value::Bool(b) => b.to_string(),
-        Value::Null => "null".to_string(),
-        // Fallback to JSON for arrays/objects
-        _ => serde_json::to_string(v).unwrap_or_default(),
+        Value::String(s) => Ok(s.clone()),
+        Value::Number(n) => Ok(n.to_string()),
+        Value::Bool(b) => Ok(b.to_string()),
+        Value::Null => Err(DscError::Parser("Array elements cannot be null".to_string())),
```

Comment:
All text needs to be in the localizable en-us.toml file,see examples using `t!()` macro